src/convertToObject.js

```diff
@@ -12,7 +12,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const filterSrc
+     = sourceString.split('\n')
+       .map(s => s.trim())
+       .filter(s => s !== '' && s !== ';');
+
+  const arrKeys = [];
+  const arrValues = [];
+  const resultObject = {};
+
+  for (let i = 0; i < filterSrc.length; i++) {
+    arrValues.push(filterSrc[i]
+      .slice(filterSrc[i].indexOf(':') + 1, filterSrc[i].indexOf(';')).trim());
+    arrKeys.push(filterSrc[i].slice(0, filterSrc[i].indexOf(':')).trim());
+
+    resultObject[arrKeys[i]] = arrValues[i];
+  }
```

Comment:
let's try to do it with array iteration methods instead of the loop. For example, with reduce

Reply:
@aionova done)

Reply:
that was fast)